flytekit/configuration/__init__.py

```diff
@@ -795,6 +795,7 @@ def new_builder(self) -> Builder:
             flytekit_virtualenv_root=self.flytekit_virtualenv_root,
             python_interpreter=self.python_interpreter,
             fast_serialization_settings=self.fast_serialization_settings,
+            source_root=self.source_root,
```

Comment:
is this change still needed?  or just for cleanliness?

Reply:
yes, we need it for dynamic task